cmake/clang-format.cmake

```diff
@@ -0,0 +1,50 @@
+# select all files with correct c++ extensions
+file(GLOB_RECURSE
+  ALL_CXX_SOURCE_FILES
+  *.[ch] *.[chi]pp *.[chi]xx *.cppm *.cc *.hh *.ii *.[CHI]
+)
+
+# exclude the docs **for now**
+set(EXCLUDE_REGEX ".*doc[\\|/].*")
+if(${CMAKE_VERSION} VERSION_LESS "3.6.0")
+  foreach(PROPOSED_FILE ${ALL_CXX_SOURCE_FILES})
+    string(REGEX MATCH ${EXCLUDE_REGEX} regex_found ${PROPOSED_FILE})
+    if(regex_found)
+      list(REMOVE_ITEM ${ALL_CXX_SOURCE_FILES} ${PROPOSED_FILE})
+    endif()
+  endforeach()
+else()
+  list(FILTER ALL_CXX_SOURCE_FILES EXCLUDE REGEX ${EXCLUDE_REGEX})
+endif()
+
+find_program(CLANG_FORMAT 
+  NAMES clang-format
+    # backup names
+    clang-format-9.0 clang-format-8.0 clang-format-7.0 clang-format-6.0
+    clang-format-9 clang-format-8 clang-format-7 clang-format-6
+)
+
+# search for version number in clang-format without version number
+if(CLANG_FORMAT)
+  execute_process(
+    COMMAND sh "-c"
```

Comment:
`sh` with pipes means that we are only Unix-compatible here. While not an issue per se (an advanced user may add support for Windows in the future), we should properly guard this. Perhaps check for Windows and exit early from this entire script.

Reply:
I was wondering how to do this on Windows. Any ideas?

Reply:
I think [this](https://github.com/BlueBrain/git-cmake-format/blob/master/FindClangFormat.cmake) works on all platforms since it only uses CMake built-in functionality.